env Screen - Wind

iamsyedomair opened the pull request.

iamsyedomair committed:
env screen wind changes

cleaned up old code by a lot. (almost half)
add button now visible on every wind panel
added a toggle to enable or disable wind selection

Anjali0407-git commented on frontend/src/components/WindSettings.jsx:
```diff
@@ -20,391 +19,209 @@ const useStyles = makeStyles((theme) => ({
+            <Grid item>
+                <FormControlLabel
+                    control={
+                        <Switch
```
@syedomair0 Remove the Switch from Wind panel. You can directly show the "+ Add Wind Source" button.

Anjali0407-git commented on frontend/src/components/WindSettings.jsx:
```diff
@@ -20,391 +19,209 @@ const useStyles = makeStyles((theme) => ({
+const WindType = [
+    { value: "Constant Wind", id: 1 },
+    { value: "Turbulent Wind", id: 2 },
+    { value: "Wind Shear", id: 3 },
```
@syedomair0 remove the Wind Shear from this menu.

Anjali0407-git commented on frontend/src/components/WindSettings.jsx:
```diff
@@ -20,391 +19,209 @@ const useStyles = makeStyles((theme) => ({
+                        {renderSelectField("Wind Direction", envConf.Wind.Direction, handleDirection, WindDirection)}
+                        {renderTextField("Wind Velocity (m/s)", envConf.Wind.Force, handleWindChange, { min: 0 })}
+
+                        {(selectedWindType === "Turbulent Wind" || selectedWindType === "Wind Shear") && (
```
@syedomair0 remove the condition for Wind Shear

Anjali0407-git commented on frontend/src/components/WindSettings.jsx:
```diff
@@ -20,391 +19,209 @@ const useStyles = makeStyles((theme) => ({
-                            ))}
-                        </StyledSelect>
+            {!isWindSelectionEnabled && (
+                <Grid item container xs={12}>
```
@syedomair0 remove the code from new line 164 to 171 as the same button is already there at new line [211](https://github.com/UAVLab-SLU/DRV_public/pull/38/files#diff-0c51e2066ebbc0f941678897fd3dbaeb0897f35ac1e80228f9bca9f1f4b48175R211)

Anjali0407-git commented on frontend/src/components/WindSettings.jsx:
```diff
@@ -20,391 +19,209 @@ const useStyles = makeStyles((theme) => ({
+                                step: 0.1
+                            })
+                        )}
                     </Grid>
```
@syedomair0 I think you can remove the above code from new line 176 to 189 as you already found a way to display multiple wind blocks using `map` function at new line 192.

Anjali0407-git commented on frontend/src/components/WindSettings.jsx:
```diff
@@ -20,391 +19,209 @@ const useStyles = makeStyles((theme) => ({
+                                setWindBlockData(index, { windDirection: e.target.value }), WindDirection)}
+                            {renderTextField("Wind Velocity (m/s)", windBlock.windVelocity, (e) =>
+                                setWindBlockData(index, { windVelocity: e.target.value }), { min: 0 })}
+                            {renderTextField("Fluctuation %", windBlock.fluctuationPercentage, (e) =>
```
@syedomair0 `Fluctuation` field should be displayed only when `Turbulent Wind` Type is selected

iamsyedomair committed:
fixes after review #1

removed switch
removed wind shear section
fixed fluctuation field to be displayed only when turbulent wind type is selected

iamsyedomair commented:
made some changes as per the review. please let me know if I missed anything.

iamsyedomair committed:
removed dead code

just noticed isWindSelectionEnabled was not needed anymore

iamsyedomair committed:
Merge branch 'frontend-updates' into omair-changes

Anjali0407-git commented on frontend/src/components/WindSettings.jsx:
```diff
@@ -19,390 +19,153 @@ const useStyles = makeStyles((theme) => ({
+        updatedWindBlocks.splice(index, 1);
+        setWindBlocks(updatedWindBlocks);
+    };
 
```
@syedomair0 Visually everything looks good to me. 
One issue is found in its functionality. Let me explain the failed test case:
 I created 3 wind blocks. When I attempted to delete 2nd block, the 3rd block has been deleted instead of 2nd.
The same occurs when I attempted to delete 3rd block from the list of 4 blocks. 4th block has been deleted instead of 3rd. 
Print the windBlocks array to console and debug this issue.

iamsyedomair replied:
thank you for pointing this out. debugged and fixed.

Anjali0407-git commented on frontend/src/components/WindSettings.jsx:
```diff
@@ -19,390 +19,153 @@ const useStyles = makeStyles((theme) => ({
+                    classes={{ root: classes.backdropFilter }}
+                    sx={{ border: '1px white solid', textAlign: 'center' }}>
+                    <IconButton onClick={addNewWindBlock} color="warning">
+                        <AddIcon />
```
@syedomair0 A small change here. Wrap the **`<AddIcon />` Add Wind Source** inside a button, so that the button label looks like **"+ Add Wind Source"**

iamsyedomair replied:
done

iamsyedomair committed:
review #2

added "+ add wind source"
added functionality to save state
fixed an edge case

iamsyedomair committed:
Merge branch 'omair-changes' of github.com:UAVLab-SLU/DRV_public into omair-changes

Anjali0407-git commented on frontend/src/components/WindSettings.jsx:
```diff
@@ -19,390 +20,159 @@ const useStyles = makeStyles((theme) => ({
+}) => {
+    const classes = useStyles();
+    const [windBlocks, setWindBlocks] = useState(() => {
+        const storedWindBlocks = localStorage.getItem('windBlocks');
```
@syedomair0 It's not a best idea to use localStorage here, because it stores values in the browser. I see that the Windblock data persists even after I refresh the webpage. 
We just have to store it in the internal state of the component.

Anjali0407-git replied:
@syedomair0 I would suggest to remove the localStorage and move the state `const [windBlocks, setWindBlocks] = useState([])` into its parent component `EnvironmentConfiguration` and pass down this state as props to child component `WindSettings`.
Also make sure to sync the **Wind** property of **envConf** in `EnvironmentConfiguration` file with the latest windBlocks data. Essentially you have to call setEnvConf(updatedWindBlocks) within setWindBlocks(updatedWindBlocks).

iamsyedomair replied:
ah yes I was thinking that too... am working on it right now. will be fixed soon.

Anjali0407-git commented on frontend/src/components/WindSettingOriginal.jsx:
```diff
@@ -0,0 +1,424 @@
+import React from 'react';
+import { Grid, TextField, IconButton, InputLabel, Tooltip, MenuItem, Button } from '@mui/material';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import { OutlinedInput } from '@mui/material';
```
@syedomair0 remove this file entirely

Anjali0407-git commented on frontend/src/components/WindSettings.jsx:
```diff
@@ -413,10 +183,8 @@ WindSettings.propTypes = {
     fluctuationPercentage: PropTypes.number.isRequired,
     windShears: PropTypes.array.isRequired,
-    addNewWindShear: PropTypes.func.isRequired,
-    windShearData: PropTypes.object.isRequired,
```
@syedomair0 Also remove anything related to Wind Shear from all files, because its all going to be dead code in new version

iamsyedomair committed:
changes after review #3

changed font size for the "add wind source" button
removed everything related to "Wind Shears"
deleted original file
now saving state locally using envConf insetad of globally
